Source/JavaScriptCore/runtime/StringPrototype.cpp

```diff
@@ -964,6 +964,25 @@ JSC_DEFINE_JIT_OPERATION(operationStringProtoFuncReplaceRegExpEmptyStr, JSCell*,
         vm, globalObject, thisValue, searchValue, callData, replacementString, JSValue()));
 }
 
+JSC_DEFINE_JIT_OPERATION(operationStringProtoFuncReplaceAllRegExpEmptyStr, JSCell*, (JSGlobalObject* globalObject, JSString* thisValue, RegExpObject* searchValue))
+{
+    VM& vm = globalObject->vm();
+    CallFrame* callFrame = DECLARE_CALL_FRAME(vm);
+    JITOperationPrologueCallFrameTracer tracer(vm, callFrame);
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    RegExp* regExp = searchValue->regExp();
+
+    ASSERT(regExp->global());
```

Comment:
Why is it guaranteed?

Reply:
Ah my misunderstood. Fixed and added tests.